### ISSUE TITLE
Problem launching IRB under Rubinius 2.5.0 on FreeBSD 10.1

### ISSUE THREAD

**Issue by dmilith** [rubinius/rubinius#3288]
```
ng0b / # irb
An exception occurred running /Software/Rubinius/library/bin/irb.rb:

No such file or directory - execvp(2) failed (Errno::ENOENT)

Backtrace:

                             Kernel(Module)#` at ernel/common/kernel.rb:188
             RbReadline.get_term_capabilities at oftware/Rubinius/gems/gems/rb-readline-0.5.2/lib/rbreadline.rb:1814
              RbReadline._rl_init_terminal_io at oftware/Rubinius/gems/gems/rb-readline-0.5.2/lib/rbreadline.rb:2026
    RbReadline.readline_initialize_everything at oftware/Rubinius/gems/gems/rb-readline-0.5.2/lib/rbreadline.rb:2512
                     RbReadline.rl_initialize at oftware/Rubinius/gems/gems/rb-readline-0.5.2/lib/rbreadline.rb:3797
                          RbReadline.readline at oftware/Rubinius/gems/gems/rb-readline-0.5.2/lib/rbreadline.rb:4816
  Readline(IRB::ReadlineInputMethod)#readline at oftware/Rubinius/gems/gems/rb-readline-0.5.2/lib/readline.rb:45
                IRB::ReadlineInputMethod#gets at oftware/Rubinius/gems/gems/rubysl-irb-2.1.0/lib/irb/input-method.rb:152
                   { } in IRB::Irb#eval_input at oftware/Rubinius/gems/gems/rubysl-irb-2.1.0/lib/rubysl/irb/irb.rb:472
                       IRB::Irb#signal_status at oftware/Rubinius/gems/gems/rubysl-irb-2.1.0/lib/rubysl/irb/irb.rb:624
                   { } in IRB::Irb#eval_input at oftware/Rubinius/gems/gems/rubysl-irb-2.1.0/lib/rubysl/irb/irb.rb:471
                                    Proc#call at ernel/bootstrap/proc.rb:20
                            RubyLex#buf_input at oftware/Rubinius/gems/gems/rubysl-irb-2.1.0/lib/irb/ruby-lex.rb:190
                                 RubyLex#getc at oftware/Rubinius/gems/gems/rubysl-irb-2.1.0/lib/irb/ruby-lex.rb:105
                     IRB::SLex::Node#match_io at oftware/Rubinius/gems/gems/rubysl-irb-2.1.0/lib/irb/slex.rb:206
                              IRB::SLex#match at oftware/Rubinius/gems/gems/rubysl-irb-2.1.0/lib/irb/slex.rb:76
                                RubyLex#token at oftware/Rubinius/gems/gems/rubysl-irb-2.1.0/lib/irb/ruby-lex.rb:290
                                  RubyLex#lex at oftware/Rubinius/gems/gems/rubysl-irb-2.1.0/lib/irb/ruby-lex.rb:266
      { } in RubyLex#each_top_level_statement at oftware/Rubinius/gems/gems/rubysl-irb-2.1.0/lib/irb/ruby-lex.rb:237
                         Kernel(RubyLex)#loop at ernel/common/kernel.rb:510
      { } in RubyLex#each_top_level_statement at oftware/Rubinius/gems/gems/rubysl-irb-2.1.0/lib/irb/ruby-lex.rb:233
                                    Proc#call at ernel/bootstrap/proc.rb:20
                 { } in Kernel(RubyLex)#catch at ernel/common/throw_catch.rb:30
               Rubinius::ThrownValue.register at ernel/common/throw_catch.rb:8
                        Kernel(RubyLex)#catch at ernel/common/throw_catch.rb:29
             RubyLex#each_top_level_statement at oftware/Rubinius/gems/gems/rubysl-irb-2.1.0/lib/irb/ruby-lex.rb:232
                          IRB::Irb#eval_input at oftware/Rubinius/gems/gems/rubysl-irb-2.1.0/lib/rubysl/irb/irb.rb:488
                             { } in IRB.start at oftware/Rubinius/gems/gems/rubysl-irb-2.1.0/lib/rubysl/irb/irb.rb:397
                                    Proc#call at ernel/bootstrap/proc.rb:20
                  { } in Kernel(Module)#catch at ernel/common/throw_catch.rb:30
               Rubinius::ThrownValue.register at ernel/common/throw_catch.rb:8
                         Kernel(Module)#catch at ernel/common/throw_catch.rb:29
                                    IRB.start at oftware/Rubinius/gems/gems/rubysl-irb-2.1.0/lib/rubysl/irb/irb.rb:396
                            Object#__script__ at oftware/Rubinius/library/bin/irb.rb:13
             Rubinius::CodeLoader#load_script at ernel/delta/code_loader.rb:66
             Rubinius::CodeLoader.load_script at ernel/delta/code_loader.rb:152
                      Rubinius::Loader#script at ernel/loader.rb:645
                        Rubinius::Loader#main at ernel/loader.rb:799
zsh: exit 1     irb
```

**Comment by dmilith** [comment]
Also happens with regular ruby:

```
ng0b / # ruby
An exception occurred running IRB:

No such file or directory - execvp(2) failed (Errno::ENOENT)

Backtrace:

                             Kernel(Module)#` at ernel/common/kernel.rb:188
             RbReadline.get_term_capabilities at oftware/Rubinius/gems/gems/rb-readline-0.5.2/lib/rbreadline.rb:1814
              RbReadline._rl_init_terminal_io at oftware/Rubinius/gems/gems/rb-readline-0.5.2/lib/rbreadline.rb:2026
    RbReadline.readline_initialize_everything at oftware/Rubinius/gems/gems/rb-readline-0.5.2/lib/rbreadline.rb:2512
                     RbReadline.rl_initialize at oftware/Rubinius/gems/gems/rb-readline-0.5.2/lib/rbreadline.rb:3797
                          RbReadline.readline at oftware/Rubinius/gems/gems/rb-readline-0.5.2/lib/rbreadline.rb:4816
  Readline(IRB::ReadlineInputMethod)#readline at oftware/Rubinius/gems/gems/rb-readline-0.5.2/lib/readline.rb:45
                IRB::ReadlineInputMethod#gets at oftware/Rubinius/gems/gems/rubysl-irb-2.1.0/lib/irb/input-method.rb:152
                   { } in IRB::Irb#eval_input at oftware/Rubinius/gems/gems/rubysl-irb-2.1.0/lib/rubysl/irb/irb.rb:472
                       IRB::Irb#signal_status at oftware/Rubinius/gems/gems/rubysl-irb-2.1.0/lib/rubysl/irb/irb.rb:624
                   { } in IRB::Irb#eval_input at oftware/Rubinius/gems/gems/rubysl-irb-2.1.0/lib/rubysl/irb/irb.rb:471
                                    Proc#call at ernel/bootstrap/proc.rb:20
                            RubyLex#buf_input at oftware/Rubinius/gems/gems/rubysl-irb-2.1.0/lib/irb/ruby-lex.rb:190
                                 RubyLex#getc at oftware/Rubinius/gems/gems/rubysl-irb-2.1.0/lib/irb/ruby-lex.rb:105
                     IRB::SLex::Node#match_io at oftware/Rubinius/gems/gems/rubysl-irb-2.1.0/lib/irb/slex.rb:206
                              IRB::SLex#match at oftware/Rubinius/gems/gems/rubysl-irb-2.1.0/lib/irb/slex.rb:76
                                RubyLex#token at oftware/Rubinius/gems/gems/rubysl-irb-2.1.0/lib/irb/ruby-lex.rb:290
                                  RubyLex#lex at oftware/Rubinius/gems/gems/rubysl-irb-2.1.0/lib/irb/ruby-lex.rb:266
      { } in RubyLex#each_top_level_statement at oftware/Rubinius/gems/gems/rubysl-irb-2.1.0/lib/irb/ruby-lex.rb:237
                         Kernel(RubyLex)#loop at ernel/common/kernel.rb:510
      { } in RubyLex#each_top_level_statement at oftware/Rubinius/gems/gems/rubysl-irb-2.1.0/lib/irb/ruby-lex.rb:233
                                    Proc#call at ernel/bootstrap/proc.rb:20
                 { } in Kernel(RubyLex)#catch at ernel/common/throw_catch.rb:30
               Rubinius::ThrownValue.register at ernel/common/throw_catch.rb:8
                        Kernel(RubyLex)#catch at ernel/common/throw_catch.rb:29
             RubyLex#each_top_level_statement at oftware/Rubinius/gems/gems/rubysl-irb-2.1.0/lib/irb/ruby-lex.rb:232
                          IRB::Irb#eval_input at oftware/Rubinius/gems/gems/rubysl-irb-2.1.0/lib/rubysl/irb/irb.rb:488
                             { } in IRB.start at oftware/Rubinius/gems/gems/rubysl-irb-2.1.0/lib/rubysl/irb/irb.rb:397
                                    Proc#call at ernel/bootstrap/proc.rb:20
                  { } in Kernel(Module)#catch at ernel/common/throw_catch.rb:30
               Rubinius::ThrownValue.register at ernel/common/throw_catch.rb:8
                         Kernel(Module)#catch at ernel/common/throw_catch.rb:29
                                    IRB.start at oftware/Rubinius/gems/gems/rubysl-irb-2.1.0/lib/rubysl/irb/irb.rb:396
                            Object#__script__ at oftware/Rubinius/gems/gems/rubysl-irb-2.1.0/bin/irb:12
                          Kernel(Object)#load at ernel/common/kernel.rb:497
                            Object#__script__ at oftware/Rubinius/gems/bin/irb:23
                Kernel(Rubinius::Loader)#load at ernel/common/kernel.rb:497
                        Rubinius::Loader#repl at ernel/loader.rb:685
                        Rubinius::Loader#main at ernel/loader.rb:800
```

**Comment by dmilith** [comment]
NOTE: didn't happen under Rubinius 2.4.1

**Comment by jemc** [comment]
Is it the same version of the rb-deadline gem?  It seems like maybe this issue should be filed there?

**Comment by dmilith** [comment]
It happens only with console.
```
gem install X
```

 works,

```
ng0b / # ruby -e "puts 'abc'"
abc
```

works

**Comment by brixen** [comment]
@dmilith do you have `infocmp` on your system? I don't have it on FreeBSD 10 patch 7.

**Comment by dmilith** [comment]
@brixen Nope

**Comment by brixen** [comment]
This should be fixed by 596937f2354f58497541935ee6a5ab941ec16927.

In the future, if rb-readline is fixed for FreeBSD, we can remove this.
